src/main/java/cs361/battleships/models/ships/Ship.java

```diff
@@ -0,0 +1,39 @@
+package cs361.battleships.models.ships;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import cs361.battleships.models.Square;
+
+import java.util.List;
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        property = "type"
+)
+
+@JsonSubTypes({
```

Comment:
A new subtype will be necessary to support Submarine.

Reply:
I appreciate the abstraction of Ship <3